### PR TITLE
fix: type checks

### PR DESCRIPTION
**Pull request by hig-dev**
Using `runtimeType` for type checks ist not recommended and I experienced a bug related to that.
This PR uses the pattern matching feature from Dart 3.0.

More info on type matching is here: https://github.com/dart-lang/language/issues/2911#issuecomment-1468272431

### CODE REVIEW

File: app/lib/screens/study/tasks/task_screen.dart [hunk 1]
```diff
@@ -34,23 +34,22 @@ class _TaskScreenState extends State<TaskScreen> {
...
         return QuestionnaireTaskWidget(
-          task: taskInstance.task as QuestionnaireTask,
+          task: questionnaireTask,
           key: UniqueKey(),
           completionPeriod: taskInstance.completionPeriod,
         );
       default:
-        print('${taskInstance.task.runtimeType} is not a supported Task!');
-        return null;
+        throw ArgumentError('Task ${taskInstance.task.type} not supported');
```
Comment (hig-dev) [inline app/lib/screens/study/tasks/task_screen.dart:52]: This default switch branch can be prevented in the future if we use sealed classes: https://dart.dev/language/class-modifiers#sealed

File: app/lib/widgets/questionnaire/question_container.dart [hunk 2]
```diff
@@ -26,46 +26,45 @@ class _QuestionContainerState extends State<QuestionContainer> with AutomaticKee
...
+      case AnnotatedScaleQuestion annotatedScaleQuestion:
         return AnnotatedScaleQuestionWidget(
-          question: widget.question as AnnotatedScaleQuestion,
+          question: annotatedScaleQuestion,
           onDone: _onDone,
         );
       default:
-        print('Question not supported!');
-        return null;
+        throw ArgumentError('Question type ${widget.question.type} not supported');
```
Comment (hig-dev) [inline app/lib/widgets/questionnaire/question_container.dart:60]: This default switch branch can be prevented in the future if we use sealed classes: https://dart.dev/language/class-modifiers#sealed

File: app/lib/screens/study/tasks/task_screen.dart (hunk 1 above)
Comment (johannesvedder) [reply app/lib/screens/study/tasks/task_screen.dart:52]: https://github.com/hpi-studyu/studyu/issues/566

File: app/lib/widgets/questionnaire/question_container.dart (hunk 2 above)
Comment (johannesvedder) [reply app/lib/widgets/questionnaire/question_container.dart:60]: https://github.com/hpi-studyu/studyu/issues/566

**Review by johannesvedder — APPROVED** [review]
Nice 👍

**Pull request merged**